### PR TITLE
Friends App

### PR DESCRIPTION
**Pull request by yaripey**
### Friends-app

[Demo](https://yaripey.github.io/friends-app/)
[Code base](https://github.com/yaripey/friends-app)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by zonzujiro** [comment]
I see common mistakes. Please check this list:

- Single responsibility - your functions should do only one job. As an example function in which you fetch users should only fetch them and not render, transform or process them in other ways. All this things should be done in another place, outside your function. Same applied to sort function, which usually do all types of sorting 😉
- You should handle fetch response status - https://www.tjvantoll.com/2015/09/13/fetch-and-errors/
- Don’t use names as arr, array, data or etc. They to generic. Don’t use things like sortingByABC and sortingByZYX - it’s unclear and just weird. `user1` and `user2` - don't use numbers in names. Just don't :) friendsArr also bad idea - you show data type in variable name and in JS it’s redundant
- Prefer template strings to “createElement” API
- Your template strings which contains should have intendations as in HTML
- When you need to decide which sort condition were selected, prefer value of input to “check class name of pressed element”, or “compare pressed element to saved element” etc. You can have value “female”, get value and then just use it
- Don’t overuse global variables, keep data which you are using closest as possible to place where you using it. As an example, request url can be inside function which sends request
- Redundant comments - don’t use stuff ‘———————————‘ or “here I will sort my array”. First one is ugly, second - redundant because reviewer will understand that you do sort, from your code. Only situation in which you should use comments, is when you need to describe why something is going on and this is 1% of all situations :)
- Please, doublecheck english wordings. Use code spellchecker, etc.

And remember, the best way to speed up review, is to fix errors prior review. 90% of errors are shared - go thru closed PRs with Friends App and check discussions there. You will find a places with same mistakes

File: submissions/yaripey/friends-app/js/userService.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import { settings } from './settings.js'
+let fetchingTries = 0
+const maxTriesAmount = 5;
+
+async function fetchUsers(amount) {
+  const usersURL = `https://randomuser.me/api/?results=${amount}`
+  try {
+    const fetchedResult = await fetch(usersURL)
+    const fetchedUsers = await fetchedResult.json()
+    console.log(fetchedUsers.results)
```
Comment (zonzujiro) [inline submissions/yaripey/friends-app/js/userService.js:11]: Please remove code that is necessary for debugging from PR :)

File: submissions/yaripey/friends-app/js/userService.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    } else {
+      console.log('Too many fetching attempts. Check connection or try again later.')
+      fetchingTries = 0;
+    }
+  }
+}
+
+export async function loadUsers(amount) {
+  const fetchedUsers = await fetchUsers(amount)
+  fetchedUsers.forEach(loadedUser => {
```
Comment (zonzujiro) [inline submissions/yaripey/friends-app/js/userService.js:27]: You will have `undefined` if all attempts will fail, no? 🤔

File: submissions/yaripey/friends-app/js/userService.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    const newUser = new User({...loadedUser, id})
+    loadedUsers.push(newUser)
+  })
+  updateUsersToShow()
+}
+
+export function peekUser({ target }) {
+  const targetElement = target.closest('.user')
+  if (targetElement) {
+    const id = target.closest('.user').id
```
Comment (zonzujiro) [inline submissions/yaripey/friends-app/js/userService.js:38]:
```suggestion
    const { id } = target.closest('.user')
```

File: submissions/yaripey/friends-app/js/userService.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      if (!(loadedUser.age === settings.ageFilter)) {
+        return false
+      }
+    }
+    return true
+  })
+  return filteredUsers;
+}
+
+function alphaSort(usersToSort, ascending) {
```
Comment (zonzujiro) [inline submissions/yaripey/friends-app/js/userService.js:72]: What with naming? :) Do you have `betaSort`?

File: submissions/yaripey/friends-app/js/userService.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    })
+  } else {
+    return usersToSort.sort(function (a, b) {
+      if (a.firstName < b.firstName) {
+        return 1;
+      }
+      if (a.firstName > b.firstName) {
+        return -1;
+      }
+      return 0;
```
Comment (zonzujiro) [inline submissions/yaripey/friends-app/js/userService.js:91]: Same function as in `ascending` ;)

File: submissions/yaripey/friends-app/js/userService.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    return usersToSort.sort(function (a, b) {
+      return b.age - a.age;
+    })
+  }
+}
+
+function sortUsers(usersToSort, sortType) {
+  switch(sortType) {
+    case 'A-Z':
+      return alphaSort(usersToSort, true);
```
Comment (zonzujiro) [inline submissions/yaripey/friends-app/js/userService.js:111]: Please, don't use `boolean` arguments. When someone else will check your code he will not understand the meaning of `true` or `false`.

Also, sort functions should understand by themself order of the sort.

**Review by zonzujiro — CHANGES_REQUESTED** [review]

File: submissions/yaripey/friends-app/js/userService.js (hunk 4 above)
Comment (yaripey) [reply submissions/yaripey/friends-app/js/userService.js:72]: It's a short for alphabetical sort :D But I see the problem here, will rename.

**Review by zonzujiro — APPROVED** [review]

**Comment by zonzujiro** [comment]
LGTM

**Pull request merged**